src/qibo/models/circuit.py

```diff
@@ -172,6 +181,35 @@ def __init__(self, nqubits, accelerators=None, density_matrix=False):
                 )
             self._distributed_init(nqubits, accelerators)
 
+        self._basis_initialization(basis, eigenstate)
+
+    def _basis_initialization(self, basis: str, eigenstate: str):
+        """This function appends some gates at the beginning of the
+        circuit's queue in order to initialize all the qubits in a specific
+        eigenstate of the operator defined in `basis`:
+
+            - if eigenstate is  '+', no gate added
+            - if eigenstate is '-', add an X gate
+            - if basis is 'Z', no gate added
+            - if basis is 'X', add a Hadamard gate
+            - if basis is 'Y', add a Hadamard and an S gate
+        """
+        initial_queue = []
+        if eigenstate == "-":
+            initial_queue.append(gates.X)
+        elif eigenstate != "+":
+            raise NotImplementedError(f"Invalid eigenstate {eigenstate}")
```

Comment:
We usually use a custom error
```suggestion
            raise_error(NotImplementedError, f"Invalid eigenstate {eigenstate}")
```
which I kind of dislike, but since the rest of code is like this it is better to not break the symmetry.